File: backend/app/schemas/locations.py
"""Location schemas."""
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class LocationBase(BaseModel):
    name: str
    parent_id: Optional[int] = None
    location_type: Optional[str] = None
    latitude: Optional[float] = None
    longitude: Optional[float] = None
    radius_meters: Optional[float] = None
    address: Optional[str] = None


class LocationCreate(LocationBase):
    pass


class LocationUpdate(BaseModel):
    name: Optional[str] = None
    parent_id: Optional[int] = None
    location_type: Optional[str] = None
    latitude: Optional[float] = None
    longitude: Optional[float] = None
    radius_meters: Optional[float] = None
    address: Optional[str] = None


class LocationResponse(LocationBase):
    model_config = ConfigDict(from_attributes=True)
    id: int
    created_at: datetime

File: frontend/src/api/locations.ts
import { apiGet, apiPost, apiPatch, apiDelete } from './client'

export interface Location {
  id: number
  name: string
  parent_id: number | null
  location_type: string | null
  latitude: number | null
  longitude: number | null
  radius_meters: number | null
  address: string | null
  created_at: string
}

export interface Paginated<T> {
  items: T[]
  total: number
  page: number
  size: number
  pages: number
}

export function getLocations(params?: { page?: number; size?: number; parent_id?: number }) {
  const searchParams: Record<string, string> = {}
  if (params?.page) searchParams.page = String(params.page)
  if (params?.size) searchParams.size = String(params.size)
  if (params?.parent_id != null) searchParams.parent_id = String(params.parent_id)
  return apiGet<Paginated<Location>>('/locations', Object.keys(searchParams).length ? searchParams : undefined)
}
